packages/astro/type-utils.d.ts

```diff
@@ -0,0 +1,16 @@
+import './astro-jsx';
+import { AstroBuiltinAttributes } from './dist/@types/astro';
+
+/** Enforce that a component accepts no props or slots */
+export type NoProps = Record<string, never>;
+/** Enforce that a component requires a default slot */
+export type WithChildren = { children: any; };
```

Comment:
Having two different patterns here make this a bit weird for me, I also don't really like that `NoProps` implicitly means `NoChildren` (even though it does make total sense internally)

I'm struggling to think of a way to make this better though, hmm

Reply:
I don't know that these two will be useful in the long-term, so I'm going to reduce our API surface and just remove these until we have a stable plan for slots.